warehouse/query-core/src/main/java/datawave/query/jexl/visitors/IteratorBuildingVisitor.java

```diff
@@ -1327,19 +1329,14 @@ public FunctionFilter(List<ASTFunctionNode> nodes) {
         
         @Override
         public boolean keep(Key k) {
-            // fieldname is after fi\0
-            String fieldName = k.getColumnFamily().toString().substring(3);
             
-            // fieldvalue is first portion of cq
-            String fieldValue = k.getColumnQualifier().toString();
-            // pull off datatype and uid
-            int index = fieldValue.lastIndexOf('\0');
-            index = fieldValue.lastIndexOf('\0', index - 1);
-            fieldValue = fieldValue.substring(0, index);
+            parser.parse(k);
+            
+            final String field = parser.getField();
+            final String value = parser.getValue();
             
-            // create a jexl context with this valud
-            JexlContext context = new DatawaveJexlContext();
-            context.set(fieldName, new ValueTuple(fieldName, fieldValue, fieldValue, null));
+            context.clear();
+            context.set(field, new ValueTuple(field, value, value, null));
```

Comment:
Consider calling `parser.clearState()` here so we don't hang on to the `Key` inside the parser now that we're done with it`

Reply:
Ahh, nevermind - parser.clearState() doesn't remove the reference to the key. Maybe we need something else that does that?